python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -828,13 +833,21 @@ def run(self, skyInfo, tempExpRefList, imageScalerList, weightList,
                                                  skyInfo.wcs,
                                                  coaddExposure.getInfo().getCoaddInputs().ccds)
 
-        for subBBox in self._subBBoxIter(skyInfo.bbox, subregionSize):
+        if self.config.doOnlineForMean and self.config.statistic == 'MEAN':
             try:
-                self.assembleSubregion(coaddExposure, subBBox, tempExpRefList, imageScalerList,
-                                       weightList, altMaskList, stats.flags, stats.ctrl,
-                                       nImage=nImage)
+                self.assembleOnlineMeanCoadd(coaddExposure, tempExpRefList, imageScalerList,
+                                             weightList, altMaskList, stats.ctrl,
+                                             nImage=nImage)
             except Exception as e:
-                self.log.fatal("Cannot compute coadd %s: %s", subBBox, e)
+                self.log.fatal("Cannot compute coadd %s", e)
```

Comment:
The error message should indicate that it tried online coaddition. Do you think it makes sense to fall back to the 'offline' coaddition if online coaddition fails, or is it going to be futile anyway?

Reply:
I think that's going to be futile, but if the user has configured things to use online coaddition I think it should do that.  Though the only real failures are somehow data not being there (which should be impossible) and that's going to be the same problem with the subregions.